app/src/main/java/org/eyeseetea/malariacare/domain/usecase/pull/PullUseCase.java

```diff
@@ -63,7 +67,11 @@ private void pullMetadata(final Callback callback) {
 
             @Override
             public void onComplete() {
-                pullData(callback);
+                if(mMetadataValidator.isValid()){
+                    pullData(callback);
+                } else {
+                    onError(new ConversionException());
```

Comment:
@idelcano Remember that domain should not know anything about conversions then has no sense that the use case creates ConversionException. I think that we should create o refactor to MetadataException where the message can be "All necessary mandatory metadata tables are not populated" for example.
Can you review it? thanks